0.88.0 conflicts with Jackson annotation processing

#### Goal
I use bunch of different libraries, including Realm and Jackson.

#### Expected Results
Project builds with Jackson annotations.

#### Actual Results

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithRealmTransformerForDevDebug'.
> java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonFormat

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:transformClassesWithRealmTransformerForDevDebug'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonFormat
	at com.android.builder.profile.Recorder$Block.handleException(Recorder.java:54)
	at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:57)
	at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:47)
	at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:173)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.doExecute(AnnotationProcessingTaskFactory.java:244)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.execute(AnnotationProcessingTaskFactory.java:231)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 14 more
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonFormat
	at javassist.CtClassType.toAnnoType(CtClassType.java:696)
	at javassist.CtClassType.toAnnotationType(CtClassType.java:586)
	at javassist.CtField.getAnnotations(CtField.java:317)
	at javassist.CtField.getAnnotations(CtField.java:290)
	at javassist.CtField$getAnnotations.call(Unknown Source)
	at io.realm.transformer.RealmTransformer$_transform_closure9$_closure16.doCall(RealmTransformer.groovy:115)
	at io.realm.transformer.RealmTransformer$_transform_closure9.doCall(RealmTransformer.groovy:114)
	at io.realm.transformer.RealmTransformer.transform(RealmTransformer.groovy:113)
	at com.android.build.api.transform.Transform.transform(Transform.java:308)
	at com.android.build.gradle.internal.pipeline.TransformTask$3.call(TransformTask.java:178)
	at com.android.build.gradle.internal.pipeline.TransformTask$3.call(TransformTask.java:174)
	at com.android.builder.profile.ThreadRecorder$1.record(ThreadRecorder.java:55)
	... 23 more


BUILD FAILED
```

#### Steps & Code to Reproduce

1. Add "com.fasterxml.jackson.core:jackson-annotations:2.7.2" to dependencies.
2. Use Jackson annotations in RealmObject subclass, e.g. JsonFormat.

```Java
// omitted getters, setters, etc
public class Dialog extends RealmObject {
    @PrimaryKey
    private int id;
    @JsonFormat(shape=JsonFormat.Shape.STRING, pattern="yyyy-MM-dd'T'HH:mm:ss'Z'", timezone="UTC")
    private Date date;
}
```
#### Version of Realm and tooling

Realm version(s): 0.88.0
Android Studio version: 2.0.0-beta6, Gradle 2.11

P.S. Works on 0.87.5

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 ### Bug fixes
 
 * Prevent throwing NullPointerException in RealmConfiguration.equals(RealmConfiguration) when RxJava is not in the classpath (#2416).
+* RealmTransformer fails because of missing annotation classes in user's project. (#2413)
 
 ## 0.88.0
 

--- a/realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy
+++ b/realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy
@@ -112,7 +112,6 @@ class RealmTransformer extends Transform {
         def allManagedFields = []
         allModelClasses.each {
             allManagedFields.addAll(it.declaredFields.findAll {
-                it.getAnnotations()
                 it.getAnnotation(Ignore.class) == null && !Modifier.isStatic(it.getModifiers())
             })
         }

